base_layer/transactions/src/transaction.rs

```diff
@@ -575,9 +581,14 @@ impl Transaction {
     /// 1. Range proofs of the outputs are valid
     ///
     /// This function does NOT check that inputs come from the UTXO set
-    pub fn validate_internal_consistency(&self, factories: &CryptoFactories) -> Result<(), TransactionError> {
-        self.body
-            .validate_internal_consistency(&self.offset, MicroTari::from(0), factories)
+    pub fn validate_internal_consistency(
+        &self,
+        factories: &CryptoFactories,
+        reward: Option<MicroTari>,
```

Comment:
Who put an option here?
Is this not just going to pollute upstream code with a none. A transaction should never have a reward?
The reward: Option<MicroTari> is pollution from the body because a block does have a reward. 


If you test a transaction with a reward, you are basically testing that the transaction does not equal 0, but equals the reward amount. I cant see a reason we would want to do this?

Reply:
Coinbase transactions do.

This was calling through to `body.validate_internal_consistency` with a hardcoded Zero. This would then always fail the consistency test with CB txs

Reply:
Changing coinbase utxo's to be transactions with 0 inputs, requires this. Makes sense now